tests/e2e-kubernetes/testsuites/credentials.go

```diff
@@ -278,6 +294,10 @@ func (t *s3CSICredentialsTestSuite) DefineTests(driver storageframework.TestDriv
 			sa := csiDriverServiceAccount(ctx, f)
 			overrideServiceAccountRole(ctx, f, sa, "")
 
+			if eksPodIdentityAgentDaemonSetForCluster(ctx, f) != nil {
```

Comment:
We should also add `IsPodMounter` check here, not sure if this would work on `SystemdMounter` tests